controllers/machineset_delete_policy.go

```diff
@@ -58,6 +58,9 @@ func oldestDeletePriority(machine *clusterv1.Machine) deletePriority {
 	if _, ok := machine.ObjectMeta.Annotations[DeleteMachineAnnotation]; ok {
 		return mustDelete
 	}
+	if machine.Status.NodeRef == nil {
```

Comment:
Not directly related to this change, but I'm wondering if we should do a better job of prioritization levels between some of these, it seems like we are setting a lot of things to mustDelete, we should probably attempt to rationalize which of these various alternatives should be prioritized in which ways.

Reply:
I thought about that and had originally set this to `betterDelete`. But then that broke a bunch of tests because the math had failed below. I tried to discuss this in slack but it's EOW so I didn't catch any attention. 🎣

Reply:
My other thought was, how or will any of these change due to Conditions?

Reply:
If I'm not wrong the three priority func  (`oldestDeletePriority`, `newestDeletePriority` and `randomDeletePolicy`) are sharing a set of "commonDeletePriority", that currently are repeated in all methods (like this change).

Should we take this opportunity to move "commonDeletePriority" so both the current change and the changes discussed above are implemented in a single point?

> we should do a better job of prioritization levels

Intuitively yes, but TBH when it comes to if there is the delete annotation or if there is a failure, I think that the answer is binary, so I'm ok with current approach unless a more specific use case is presented 

> will these change due to Conditions

At this stage, IMO we should not create dependendences between the two changes

Reply:
Let's open an issue and discuss there for v0.4, this is a redesign of MachineSet priorities that might be interesting for other uses as well

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3227